node-drainer/pkg/initializer/init.go

```diff
@@ -78,33 +95,38 @@ func InitializeAll(ctx context.Context, params InitializationParams) (*Component
 	}
 
 	stateManager := initializeStateManager(clientSet)
-	reconcilerCfg := createReconcilerConfig(*tomlCfg, mongoConfig, tokenConfig, pipeline, stateManager)
+	reconcilerCfg := createReconcilerConfig(*tomlCfg, databaseConfig, tokenConfig, pipeline, stateManager)
 
-	// Reconciler creates its own queue manager
-	reconciler := initializeReconciler(reconcilerCfg, params.DryRun, clientSet, informersInstance)
-	queueManager := reconciler.GetQueueManager()
+	// Create client factory and database client
+	clientFactory := factory.NewClientFactory(databaseConfig)
 
-	collection, err := initializeMongoCollection(ctx, mongoConfig)
+	// Create the database client for the reconciler to use for preprocessing
+	databaseClient, err := clientFactory.CreateDatabaseClient(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("error while initializing mongo collection: %w", err)
+		return nil, fmt.Errorf("failed to create database client: %w", err)
 	}
 
-	eventWatcher := mongodb.NewEventWatcher(
-		mongoConfig,
-		tokenConfig,
-		pipeline,
-		queueManager,
-		collection,
-	)
+	// Reconciler creates its own queue manager and needs the database client
+	reconciler := initializeReconciler(reconcilerCfg, params.DryRun, clientSet, informersInstance, databaseClient)
+	queueManager := reconciler.GetQueueManager()
+
+	// CRITICAL: Pass the existing databaseClient to avoid creating duplicate clients
+	changeStreamWatcher, err := clientFactory.CreateChangeStreamWatcher(ctx, databaseClient, "node-drainer", pipeline)
```

Comment:
So this is what the other comment was kinda talking about. I personal prefer setting up the `databaseClient` and `changeStreamWatcher` in `InitializeAll`. The other modules/controllers set it up in `reconciler.Start()` either way works but it would just be good to have consistency across the different modules for how they configure the things